packages/firebase_messaging/android/src/main/java/io/flutter/plugins/firebasemessaging/FlutterFirebaseMessagingUtils.java

```diff
@@ -0,0 +1,58 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugins.firebasemessaging;
+
+import android.app.ActivityManager;
+import android.app.KeyguardManager;
+import android.content.Context;
+import android.content.pm.ApplicationInfo;
+import android.content.pm.PackageManager;
+import android.os.Bundle;
+import android.os.Process;
+import java.util.List;
+
+public class FlutterFirebaseMessagingUtils {
+
+  private static final String DATA_MESSAGES_KEY = "flutter.firebase.data.message";
+
+  static boolean isDataMessages(Context context) {
```

Comment:
I found this method name kind of confusing. I think it might make sense to call it `dataMessagesEnabled` since you're intending this as an app-wide flag. Would it make sense to just do something by default? React Native Firebase doesn't seem to have this flag so I'm wondering how the approaches compare.